### ISSUE TITLE
bug: The autofocus property has no effect on ion-input within modal

### ISSUE THREAD

**Issue by MateusCampoSantos** [ionic-team/ionic-framework#24009]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

Currently, I'm using the autofocus property so that the input inside the modal is highlighted, but it doesn't happen that way.

`<ion-input
    autofocus="true"
    id="scannerInput"
    placeholder="Leia o QRCode"
  ></ion-input>`

### Expected Behavior

When clicking on the button that activates the modal, I would like the input that it contains to be automatically brought into focus.

### Steps to Reproduce

1. Insert `<ion-input autofocus="true"></ion-input>` in modal
2. Run ionic serve in Ionic CLI
3. Wait for the browser to load, activate the modal and observe the behavior.

### Ionic Info

```
Ionic:

   Ionic Framework               : @ionic/angular  5.6.0
   @angular-devkit/build-angular : 0.1102.3
   @angular-devkit/schematics    : 11.2.3
   @angular/cli                  : 11.2.3
   @ionic/angular-toolkit        : 3.1.0

System:

   NodeJS : v14.18.0
   npm    : 6.14.15
   OS     : Ubuntu 20.04
```

**Comment by bogomips** [comment]
In my case, the autofocus itself works, but the keyboard does not show up. 
The same code in a page triggers the keyboard to appear.
Any suggestion?

**Comment by tetsuwanadam** [comment]
I just tried autofocus on an ion-input in a modal on iOS and found it works the first time (albeit with the cursor moving a little strangely as the modal opens), but if I close the modal and reopen it autofocus no longer works. Reset the app and it works again the first time only.

**Comment by JulienLecoq** [comment]
Same issue here on Ionic 7 on my iPhone as described by @tetsuwanadam.

**Comment by liamdebeasi** [comment]
Hi everyone,

The team recently did some investigation into the reliability of `autofocus` in other components such as `ion-input`. We found that mobile platforms apply various restrictions to how `autofocus` works to the point that it is pretty unreliable in mobile applications.

We've created new documentation that shows how to manage focus in a mobile application: https://ionicframework.com/docs/developing/managing-focus This can be applied to `ion-input` to get the desired results. I am going to close this, but let me know if there are any questions.
